app/src/main/java/com/github/sdp/mediato/data/UserDatabase.java

```diff
@@ -305,9 +305,12 @@ public static CompletableFuture<List<String>> getNearbyUsernames(String username
         List<String> nearbyUsers = new ArrayList<>();
         getSavedLocation(username).thenAccept(
                 location -> {
-                    if (!location.isValid())
-                        future.completeExceptionally(new Exception("we don't have the current user's location on the database"));
+                    if (!location.isValid()) {
+                        System.out.println("No location, fetching all usernames");

```

Comment:
It would be better to use the logger here instead of print out.

Reply:
Done